Sources/Swifter/Socket.swift

```diff
@@ -51,27 +51,29 @@ public class Socket: Hashable, Equatable {
     
     public func port() throws -> in_port_t {
         var addr = sockaddr_in()
+        let addr_copy = addr
         return try withUnsafePointer(to: &addr) { pointer in
             var len = socklen_t(MemoryLayout<sockaddr_in>.size)
             if getsockname(socketFileDescriptor, UnsafeMutablePointer(OpaquePointer(pointer)), &len) != 0 {
                 throw SocketError.getSockNameFailed(Process.lastErrno)
             }
             #if os(Linux)
-                return ntohs(addr.sin_port)
```

Comment:
The other way I thought of doing this was creating the immutable let just before this `#if`.